src/main/java/gov/nist/oar/rmm/repositories/impl/VersionReleasesetsRepositoryImpl.java

```diff
@@ -0,0 +1,157 @@
+package gov.nist.oar.rmm.repositories.impl;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import org.bson.Document;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.domain.Pageable;
+import org.springframework.stereotype.Service;
+import org.springframework.util.MultiValueMap;
+
+import com.mongodb.client.AggregateIterable;
+import com.mongodb.client.MongoCollection;
+import com.mongodb.client.model.Filters;
+
+import gov.nist.oar.rmm.config.AppConfig;
+import gov.nist.oar.rmm.config.MongoConfig;
+import gov.nist.oar.rmm.exceptions.ResourceNotFoundException;
+import gov.nist.oar.rmm.repositories.VersionReleasesetsRepository;
+import gov.nist.oar.rmm.utilities.ProcessRequest;
+
+@Service
+public class VersionReleasesetsRepositoryImpl implements VersionReleasesetsRepository {
+
+    private Logger logger = LoggerFactory.getLogger(VersionReleasesetsRepositoryImpl.class);
+
+    @Autowired
+    MongoConfig mconfig;
+
+    @Autowired
+    AppConfig appconfig;
+
+    /**
+     * Search Versions dataset to get version of the record This can be queried
+     * using key,value pairs as parameters
+     * 
+     * @param param key,value parameters
+     * @param p     pagination
+     * @return
+     */
+    @Override
+    public Document findVersion(MultiValueMap<String, String> param, Pageable p) {
+
+        return this.queryResults(param, p, "Versions");
+    }
+
+    /**
+     * Search Releasesets dataset using key,value pair where key is any field in the
+     * Releasets metadata schema
+     * 
+     * @param param
+     * @param p
+     * @return
+     */
+    @Override
+    public Document findReleaseset(MultiValueMap<String, String> param, Pageable p) {
+        return this.queryResults(param, p, "ReleaseSets");
+    }
+
+    /**
+     * This is a common utility used by the findVersions and findReleasesets. This
+     * helps query given collection with specified criteria.
+     * 
+     * @param param          key,value pairs
+     * @param p              pagination information
+     * @param collectionName Name of the dataset to search from
+     * @return
+     */
+    private Document queryResults(MultiValueMap<String, String> param, Pageable p, String collectionName) {
+        ProcessRequest request = new ProcessRequest();
+        request.parseSearch(param);
+        MongoCollection<Document> mcollection = null;
+        if ("versions".equalsIgnoreCase(collectionName))
+            mcollection = mconfig.getVersionsCollection();
+        else if ("releaseSets".equalsIgnoreCase(collectionName))
```

Comment:
This hard-coded collection name does not match the hard-coded name in line 77. Use instead the parameterized value from `MongoConfig`, `mconfig.releasesets`.   Do the same for versions in line 75.